utils/filesystem/files.go

```diff
@@ -695,6 +695,23 @@ func (fs *VFS) Chown(name string, uid, gid int) (err error) {
 	return
 }
 
+func (fs *VFS) FetchOwners(name string) (uid, gid int, err error) {
+	stat, err := fs.Stat(name)
+	if err != nil {
+		return
+	}
+	if stat == nil {
+		err = fmt.Errorf("%w: missing file info", commonerrors.ErrUndefined)
+		return
+	}
+	if reflection.IsEmpty(stat.Sys()) {
+		err = commonerrors.ErrNotImplemented
```

Comment:
ErrNotImplemented feels like the wrong error since .Sys() is calling the underlying OS stat function so that should never be not implemented? It feels like ErrUnexepected or ErrUndefined might be better, but this is a minor nitpick and it may be me misunderstanding .Sys() (i.e. you know it is being used somewhere where there it will be unavialable)

Reply:
Sys() actually may not be implemented on some platforms or for some filesystems : For instance, https://github.com/spf13/afero/blob/2a70f2bb2db1524bf2aa3ca0cfebefa8d6367b7b/mem/file.go#L339. Hence, why this check is in place